### PR TITLE
Web/improvement/label mark as read

### LINKED ISSUE #180: Make mark all as read button clearer
It needs to be a text label or something - seems a lot of people don't know it's there.

image

### PR DESCRIPTION
**Pull request by bakeiro**

### CODE REVIEW

**Comment by bakeiro** [comment]
do not merge label since I need to adapt some tests, just created the PR for previewing the commit changes

**Comment by bakeiro** [comment]
Formatted my code with make format
- [x] There are no warnings from make lint
- [x] There are no console warnings when running the app
- [ ] Added any new components to storybook // Just modifying existing component
- [ ] Added Modified tests where relevant // using existing tests
- [x] All tests pass
- [x] Clicked around my changes running locally and it works
- [x] Checked Desktop, Mobile and Tablet screen sizes

**Comment by bakeiro** [comment]
now the button is above the tabs, the reason behind it, is that the button with the new label next to the tabs makes the mobile/tablet view broken (since it's to long) so I just moved above the tabs since don't look that bad (IMHO)

preview:
image
image


feel free to give any feedback about this change :)

File: features/messages/requests/MarkAllReadButton.tsx
Comment (bakeiro) [inline features/messages/requests/MarkAllReadButton.tsx:79]: why this change?
arial label isn't visible anymore to the user, so I just needed to change the text so the current tests work properly

Comment (darrenvong) [reply features/messages/requests/MarkAllReadButton.tsx:79]: Since you moved the label out (it's not hidden anymore like you said), you can remove this label altogether. Otherwise, screen readers will read this out twice. For non-English users (eventually when we turn translations on), they'll hear both languages side-by-side (since you've now hardcoded the English message here) which will be a very confusing experience

File: features/messages/requests/MarkAllReadButton.tsx
Comment (darrenvong) [inline features/messages/requests/MarkAllReadButton.tsx:81]:
```suggestion
```
We don't want the button to be permanently disabled, since the button won't ever be focusable otherwise.

**Review by darrenvong — CHANGES_REQUESTED** [review]
Might it be worth double checking with Dom from design? Otherwise, looks fine to me too!

File: features/messages/requests/MarkAllReadButton.tsx
Comment (darrenvong) [reply features/messages/requests/MarkAllReadButton.tsx:79]: Ahh I see the it's because `IconButton` requires an aria-label. You can use the `Button` component we have (instead of importing directly from `@material-ui/core`) and put the checkmark icon together with the text label as direct children, as it's not necessary to have a button nested inside a button (and it's invalid HTML).

**Comment by bakeiro** [comment]
design update (using the new Button component looks a bit different)
normal | with focus | loading

image
image
image

File: features/messages/requests/MarkAllReadButton.tsx
Comment (darrenvong) [inline features/messages/requests/MarkAllReadButton.tsx:77]: Probably a small nitpick, but I think it might be worth wrapping the text around a Typography and add a tiny bit of padding between the icon and the text

File: features/messages/requests/MarkAllReadButton.tsx
Comment (darrenvong) [inline features/messages/requests/MarkAllReadButton.tsx:1]:
```suggestion
import Button from "components/Button";
```
The folder has a default import in `index`

**Review by darrenvong — CHANGES_REQUESTED** [review]
Some small things, but otherwise let's get it in!

File: features/messages/requests/MarkAllReadButton.tsx
Comment (dommers-ux) [reply features/messages/requests/MarkAllReadButton.tsx:77]: I agree with @darrenvong ! 

Screen Shot 2022-09-19 at 12 17 20 PM

**Comment by dommers-ux** [comment]
This is a nice clean up solution for now (good job!). Below is a modified version of what you created. As Darren said before, just giving it some padding (16px) and a border to make it a visible button is a better UI. 

Screen Shot 2022-09-19 at 12 17 20 PM

--------

In the future, we would implement a similar model to this one: 

Screen Shot 2022-09-19 at 12 20 10 PM


By having a drop-down icon available, it adds some flexibility later (being able to copy it for 1v1 or group chats instead of the whole inbox at once). It will also give better flexibility later on to what other options/features can be added (block, mark as unread, etc). 

Cheers!

**Comment by bakeiro** [comment]
updated the button styles

image
image

**Review by bakeiro — COMMENTED** [review]
modified the button styles as mentioned, ready? (also, not sure how to mark conversations as resolved, if something pending please tell me :D @darrenvong

**Review by bakeiro — COMMENTED** [review]
..

**Review by darrenvong — CHANGES_REQUESTED** [review]
Hopefully should be the last things!

FYI, you can re-request a review once you're done by clicking the arrow next to my name to bust the "request change" status. And you should be able to resolve each comment with the "resolve conversation" button on each, presumably like you did on one of them

**Comment by bakeiro** [comment]
done last things + small margin bottom at the end of the button, sorry for the late reply!

**Review by darrenvong — APPROVED** [review]
LGTM, sorry for the delay in re-reviewing too!

**Comment by darrenvong** [comment]
Usually I'd leave you the honour to click the merge button, but since I rebased to keep the branch up to date (to stop the GitHub warning and avoid another round of back and forth), I thought I'd do it this time while I'm on it 😅

**Pull request merged**
